src/webgpu/shader/execution/expression/binary/af_data.ts

```diff
@@ -0,0 +1,286 @@
+import { crc32 } from '../../../../../common/util/crc32.js';
+import { kValue } from '../../../../util/constants.js';
+import { FP, FPInterval } from '../../../../util/floating_point.js';
+import { sparseScalarF64Range } from '../../../../util/math.js';
+import { reinterpretU64AsF64 } from '../../../../util/reinterpret.js';
+
+/** Line format af_data_gen/main.cpp outputs */
+interface RawBinaryData {
+  lhs: number;
+  rhs: number;
+  expected: number[];
```

Comment:
Should add a few `readonly`s and `as const`s to make sure there's no mutable global state in this file. Suggestions here and below.

```suggestion
  readonly lhs: number;
  readonly rhs: number;
  readonly expected: readonly number[];
```